Fix parsing configuration for TraceIdRatioBasedSampler on non standard locales + fix localization related test (#6444)
open-telemetry/opentelemetry-dotnet: src/OpenTelemetry/CHANGELOG.md

  }
  ```

> * When using the `OTEL_TRACES_SAMPLER_ARG` environment variable

Reviewer: ```suggestion
* Fixed parsing of `OTEL_TRACES_SAMPLER_ARG` decimal values to always use `.` as the delimiter when using the `traceidratio` sampler, preventing locale-specific parsing issues.
```

  }
  ```

> 
> * When using the `OTEL_TRACES_SAMPLER_ARG` variable with the
>   `OTEL_TRACES_SAMPLER` variable set to `traceidratio`,  the value will be
>   parsed invariantly of the culture of the local machine. Only `.` can now
>   be used as the decimal delimiter for its value.
>   ([#6444](https://github.com/open-telemetry/opentelemetry-dotnet/pull/6444))

Reviewer 2: ```suggestion

* When using the `OTEL_TRACES_SAMPLER_ARG` environment variable
  with the `OTEL_TRACES_SAMPLER` environment variable set to
  `traceidratio`,  the value will be parsed using `.` as the decimal delimiter.
  ([#6444](https://github.com/open-telemetry/opentelemetry-dotnet/pull/6444))
```

  }
  ```

> 

Reviewer 3: ```suggestion
```

  }
  ```

> * When using the `OTEL_TRACES_SAMPLER_ARG` variable with the `OTEL_TRACES_SAMPLER` variable set to `traceidratio`, the value will be parsed invariantly of the culture of the local machine. Only `.` can now be used as the decimal delimiter for its value.

Reviewer 3: ```suggestion

* When using the `OTEL_TRACES_SAMPLER_ARG` variable with the
  `OTEL_TRACES_SAMPLER` variable set to `traceidratio`,  the value will be
  parsed invariantly of the culture of the local machine. Only `.` can now
  be used as the decimal delimiter for its value.
  ([#6444](https://github.com/open-telemetry/opentelemetry-dotnet/pull/6444))
```
